src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,80 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { getUser } from '../../api';
+import { useAppDispatch } from '../../app/hooks';
+import { actions } from '../../features/currentTodo';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+type Props = {
+  todo: Todo;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    getUser(todo?.userId)
+      .then(data => {
+        setUser(data);
+      })
+      .catch(err => {
+        throw err;
+      });
+  }, []);
+
+  const closeModal = () => {
+    dispatch(actions.removeTodo());
+  };
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
+      {!user ? (
+        <Loader />
+      ) : (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              {`Todo #${todo.id}`}
+            </div>
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={closeModal}
+              area-label
```

Comment:
This is a label for screen readers, you need to provide text there
```suggestion
              aria-label="Close"
```